### DISCUSSION TITLE
undefined symbol in psql pgroonga.so

### DISCUSSION THREAD

**Discussion by EazyAhmed** [pgroonga/pgroonga#417] [Q&A] [answered]
I have installed pgroonga ang groonga both from source. Using postgres 16.2 installed from source as well. 
When i run create extension command i get. 
`ERROR:  could not load library "/usr/local/pgsql/lib/pgroonga.so": /usr/local/pgsql/lib/pgroonga.so: undefined symbol: grn_progress_index_get_n_processed_records
`

**Comment by askdkc** [comment, marked as answer]
How about doing

```bash
vi /etc/ld.so.conf.d/groonga.conf 
```
And add `/usr/local/pgsql/lib`

Then run
```bash
sudo ldconfig
```

Does that fix your problem?

**Reply by EazyAhmed** [reply]
It worked. Mind explaining why it worked? psql had the location of the so file. why is this entry making a difference?

**Reply by askdkc** [reply]
Sometimes, especially manually compile and make install some software, your machine doesn't know where to look for some shared libraries it requires.

Using `ldconfig` tells your machine where to look at.

**Reply by EazyAhmed** [reply]
psql had the correct location of the .so file. So it did know the location of the .so didnt it.

**Reply by askdkc** [reply]
as the conf name suggests, it's groonga.

**Comment by kou** [comment]
Could you show the output of `ldd /usr/local/pgsql/lib/pgroonga.so`?

**Reply by EazyAhmed** [reply]
```text
	linux-vdso.so.1 (0x00007ffd05f81000)
	libgroonga.so.0 => /usr/local/lib/libgroonga.so.0 (0x00007f32a8c00000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f32a8800000)
	libmsgpackc.so.2 => /lib/x86_64-linux-gnu/libmsgpackc.so.2 (0x00007f32a9bb0000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f32a9b94000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f32a8400000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f32a8b19000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f32a9b72000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f32a9c2f000)
```

**Reply by kou** [reply]
Did you install Groonga to `/usr/local/` not `/usr/local/pgsql/`, right?
Then, you can remove `/etc/ld.so.conf.d/groonga.conf`. You just needed to run `sudo ldconfig`.
In general, many platforms have `/usr/local/lib/` in the default dynamic linker search path. (You may have `/etc/ld.so.conf.d/libc.conf` and it may include `/usr/local/lib`.)

You need to re-run `sudo ldconfig` when you add a new file to dynamic linker search path. (In this case, you added `libgroonga.so` to `/usr/local/lib/`.)
